core/app/models/spree/order/checkout.rb

```diff
@@ -1,338 +1,338 @@
 module Spree
   class Order < Spree::Base
     module Checkout
-      extend ActiveSupport::Concern
-
-      included do
-        class_attribute :next_event_transitions
-        class_attribute :previous_states
-        class_attribute :checkout_flow
-        class_attribute :checkout_steps
-        class_attribute :removed_transitions
-
-        self.checkout_steps ||= {}
-        self.next_event_transitions ||= []
-        self.previous_states ||= [:cart]
-        self.removed_transitions ||= []
-
-        def self.checkout_flow(&block)
-          if block_given?
-            @checkout_flow = block
-            define_state_machine!
-          else
-            @checkout_flow
+      def self.included(klass)
+        klass.class_eval do
+          class_attribute :next_event_transitions
+          class_attribute :previous_states
+          class_attribute :checkout_flow
+          class_attribute :checkout_steps
+          class_attribute :removed_transitions
+
+          self.checkout_steps ||= {}
+          self.next_event_transitions ||= []
+          self.previous_states ||= [:cart]
+          self.removed_transitions ||= []
+
+          def self.checkout_flow(&block)
+            if block_given?
+              @checkout_flow = block
+              define_state_machine!
+            else
+              @checkout_flow
+            end
           end
-        end
-
-        def self.define_state_machine!
-          self.checkout_steps = {}
-          self.next_event_transitions = []
-          self.previous_states = [:cart]
-          self.removed_transitions = []
 
-          # Build the checkout flow using the checkout_flow defined either
-          # within the Order class, or a decorator for that class.
-          #
-          # This method may be called multiple times depending on if the
-          # checkout_flow is re-defined in a decorator or not.
-          instance_eval(&checkout_flow)
-
-          klass = self
-
-          # To avoid a ton of warnings when the state machine is re-defined
-          StateMachines::Machine.ignore_method_conflicts = true
-          # To avoid multiple occurrences of the same transition being defined
-          # On first definition, state_machines will not be defined
-          state_machines.clear if respond_to?(:state_machines)
-          state_machine :state, initial: :cart, use_transactions: false, action: :save_state do
-            klass.next_event_transitions.each { |t| transition(t.merge(on: :next)) }
-
-            # Persist the state on the order
-            after_transition do |order, transition|
-              order.state = order.state
-              order.state_changes.create(
-                previous_state: transition.from,
-                next_state:     transition.to,
-                name:           'order',
-                user_id:        order.user_id
-              )
-              order.save
-            end
+          def self.define_state_machine!
```

Comment:
Assignment Branch Condition size for define_state_machine! is too high. [75.33/15]